pkg/webhook/validate.go

```diff
@@ -150,12 +150,19 @@ func staticValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePol
 	// To configure RoCE on baremetal or virtual machine:
 	// BM: DeviceType = netdevice && isRdma = true
 	// VM: DeviceType = vfio-pci && isRdma = false
-	if cr.Spec.DeviceType == "vfio-pci" && cr.Spec.IsRdma {
+	if cr.Spec.DeviceType == constants.DeviceTypeVfioPci && cr.Spec.IsRdma {
 		return false, fmt.Errorf("'deviceType: vfio-pci' conflicts with 'isRdma: true'; Set 'deviceType' to (string)'netdevice' Or Set 'isRdma' to (bool)'false'")
 	}
 	if strings.EqualFold(cr.Spec.LinkType, constants.LinkTypeIB) && !cr.Spec.IsRdma {
 		return false, fmt.Errorf("'linkType: ib or IB' requires 'isRdma: true'; Set 'isRdma' to (bool)'true'")
 	}
+	// vdpa specific validation
+	if cr.Spec.VdpaType == constants.VdpaTypeVHost {
+		return false, fmt.Errorf("'vdpaType: vhost' is not supported yet")
+	}
+	if cr.Spec.DeviceType == constants.DeviceTypeVfioPci && cr.Spec.VdpaType == constants.VdpaTypeVirtio {
```

Comment:
another point today we only support the MLX device should we also check that here?

Reply:
Ok, will do